File: plugins/base/app/commons/ErrorCodes.java
package commons;

import org.apache.commons.lang3.StringUtils;

/**
 * Created by zhangmeng on 16-7-25.
 */
public class ErrorCodes {

    //系统级别
    public static ErrorCode SUCCESS = new ErrorCode(0, "SUCCESS");
    public static ErrorCode BAD_REQUEST = new ErrorCode(400, "参数验证失败");
    public static ErrorCode BUSIZ_FAILURE = new ErrorCode(404, "业务失败");
    public static ErrorCode SYSTEM_ERROR = new ErrorCode(500, "系统异常");


    public static ErrorCode with(Exception e){
        if(StringUtils.isNotBlank(e.getMessage())){
            return new ErrorCode(10000, e.getMessage());
        }else {
            return SYSTEM_ERROR;
        }
    }

}

File: plugins/base/app/commons/ErrorCodes.scala
package commons

/**
 * Created by zhangmeng on 16-9-1.
 */
case class ErrorCode(status: Int, message: String)

object ErrorCodes {
  var SUCCESS: ErrorCode = new ErrorCode(Integer.valueOf(0), "SUCCESS")
  var BAD_REQUEST: ErrorCode = new ErrorCode(Integer.valueOf(400), "参数验证失败")
  var ENTITY_NOTFOUND: ErrorCode = new ErrorCode(Integer.valueOf(401), "实体不存在")
  var PARAMS_JSON_INVALID: ErrorCode = new ErrorCode(Integer.valueOf(403), "JSON参数无法解析")
  var BUSIZ_FAILURE: ErrorCode = new ErrorCode(Integer.valueOf(404), "业务失败")
  var SYSTEM_ERROR: ErrorCode = new ErrorCode(Integer.valueOf(500), "系统异常")
}
